stetho/src/main/java/com/facebook/stetho/inspector/network/AsyncPrettyPrinterExecutorHolder.java

```diff
@@ -16,5 +16,21 @@
  * A holder class for the executor service used for pretty printing related tasks
  */
 class AsyncPrettyPrinterExecutorHolder {
-  public static final ExecutorService sExecutorService = Executors.newCachedThreadPool();
+  private static ExecutorService sExecutorService;
+
+  static void ensureInitialized() {
+    if (sExecutorService == null) {
+      sExecutorService = Executors.newCachedThreadPool();
+    }
+  }
+
+  public static ExecutorService getExecutorService() {
```

Comment:
It actually technically can be null. There is no synchronization on `sExecutorService`, and thus someone can call `shutdown` immediatley after the call to `ensureInitialized()` This will result in `sExecutorService` being null and crashing the app where it is used.

Additionally the correct thing to do, given the current design, is to not call `ensureInitialized` here for the case I mentioned above. If the last peer has disconnected you do not want to re-initialize the executor.